arlo-client/src/components/Audit/Setup/useAuditSettings.tsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useCallback, useState } from 'react'
+import { api, checkAndToast } from '../../utilities'
+import { IAuditSettings, IErrorResponse } from '../../../types'
+
+const defaultValues: IAuditSettings = {
+  state: null,
+  electionName: null,
+  online: null,
+  randomSeed: null,
+  riskLimit: null,
+}
+
+type TNewSettings =
+  | {
+      state: IAuditSettings['state']
+    }
+  | {
+      electionName: IAuditSettings['electionName']
+      online: IAuditSettings['online']
+      randomSeed: IAuditSettings['randomSeed']
+      riskLimit: IAuditSettings['riskLimit']
+    }
+
+const useAuditSettings = (
+  electionId: string
+): [IAuditSettings, (arg0: TNewSettings) => Promise<boolean>] => {
+  const [settings, setSettings] = useState(defaultValues)
+
+  const getSettings = useCallback(async (): Promise<IAuditSettings> => {
+    const settingsOrError: IAuditSettings | IErrorResponse = await api(
+      `/election/${electionId}/settings`
+    )
+    if (checkAndToast(settingsOrError)) {
+      return defaultValues
+    }
+    return settingsOrError
+  }, [electionId])
+
+  const updateSettings = async (
+    newSettings: TNewSettings
+  ): Promise<boolean> => {
+    const oldSettings = await getSettings()
+    const mergedSettings = {
+      ...oldSettings,
+      ...newSettings,
+    }
+    const response: IErrorResponse = await api(
+      `/election/${electionId}/settings`,
+      {
+        method: 'PUT',
+        body: JSON.stringify(mergedSettings),
+        headers: {
+          'Content-Type': 'application/json',
+        },
```

Comment:
Would it be useful to make a helper function to send json to the API? The server doesn't have good error messages if you forget this header (as we learned a few days ago).

Reply:
We could possibly have a couple variations of the `api` helper for sending different kinds of requests.

Reply:
We could possibly have a couple variations of the `api` helper for sending different kinds of requests.